allofplos/corpus_helpers.py

```diff
@@ -0,0 +1,95 @@
+
+    return file_hash
+
+
+def search_solr_records(days_ago=14, start=0, rows=1000, start_date=None, end_date=None, item='id'):
+    """
+    Queries the solr database for a list of articles based on the date of publication
+    function defaults to querying by DOI (i.e., 'id')
+    TODO (on hold): if Solr XSLT is changed, query by revision_date instead of publication_date.
+    Then would be used in separate query to figure out updated articles to download
+    for full list of potential queries, see http://api.plos.org/solr/search-fields/
+    :param days_ago: A int value with the length of the queried date range, default is two weeks
+    :param start: An int value indicating the first row of results to return
+    :param rows: An int value indicating how many rows of results to return (from 0)
+    :param start_date: datetime object of earliest date in the queried range (defaults to None)
+    :param end_date: datetime object of latest date in the queried range (defaults to now)
+    :param item: Items to return/display. 'Id', the default, is the article DOI.
+    :return: A list of DOIs for articles published in this time period; by default, from the last two weeks
+    """
+    if end_date is None:
+        end_date = datetime.datetime.now()
+    solr_search_results = []
+    if start_date is None:
+        earlier = datetime.timedelta(days=days_ago)
+        start_date = end_date - earlier
+    START_DATE = start_date.strftime("%Y-%m-%d")
+    END_DATE = end_date.strftime("%Y-%m-%d")
+    howmanyarticles_url_base = [BASE_URL_API,
+                                '?q=*:*&fq=doc_type:full+-doi:image&fl=id,',
+                                item,
+                                '&wt=json&indent=true&sort=%20id%20asc&fq=publication_date:[',
+                                START_DATE,
+                                'T00:00:00Z+TO+',
+                                END_DATE,
+                                'T23:59:59Z]'
+                                ]
+    howmanyarticles_url = ''.join(howmanyarticles_url_base) + '&rows=1000'
+    # if include_uncorrected is False:
+    num_results = requests.get(howmanyarticles_url).json()["response"]["numFound"]
+
+    # Create solr_search_results & paginate through results
+    while(start < num_results):
+        query_url = ''.join(howmanyarticles_url_base) + '&start=' + str(start) + '&rows=' + str(rows)
+        article_search = requests.get(query_url).json()
+        solr_search_results = [x[item] for x in article_search["response"]["docs"]]
+        start = start + rows
+        if start + rows > num_results:
+            rows = num_results - start
+    print("URL for solr query:", howmanyarticles_url)
+
+    if solr_search_results:
+        print("{0} results returned from this search."
+              .format(len(solr_search_results)))
+    else:
+        print('No results returned for this search.')
+    return solr_search_results
+
+
+def get_all_solr_dois():
```

Comment:
This should be in the same utility PR as `search_solr_records`. This seems wholly unrelated to the Corpus class.